Datagrid/ProxyQuery.php

```diff
@@ -132,10 +133,17 @@ protected function getFixedQueryBuilder(QueryBuilder $queryBuilder)
         }
 
         $results    = $queryBuilderId->getQuery()->execute(array(), Query::HYDRATE_ARRAY);
+        $platform = $queryBuilderId->getEntityManager()->getConnection()->getDatabasePlatform();
         $idxMatrix  = array();
         foreach ($results as $id) {
             foreach ($idNames as $idName) {
-                $idxMatrix[$idName][] = $id[$idName];
+                $phpValue = $id[$idName];
+                // Convert ids to database value in case of custom type
+                $fieldType = $metadata->getTypeOfField($idName);
+                $type = Type::getType($fieldType);
+                $dbValue = $type->convertToDatabaseValue($phpValue, $platform);
```

Comment:
Doctrine will return an object for the identifier (in my case an object of type Uuid). But further on in the code it expects a string value to display and use in urls. Therefor getting the db value solves to problem. It's the code from PR #481 but i've added the missing tests and some code cleanup.